### PR TITLE
Migrate to Gradle Kotlin DSL, version catalog and ksp

### PR DESCRIPTION
**Pull request by aualbert**
Closes https://github.com/Ashinch/ReadYou/issues/720.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+buildscript {
+    dependencies {
+        classpath(libs.daggerHiltPlugin)
+    }
+}
+
+plugins {
+    id("com.android.application") version libs.versions.androidGradlePlugin.get() apply false
```
Comment (JunkFood02) [inline build.gradle.kts:8]: Plugins can be also declared in version catalogs, then we can use it like this:

```
alias(libs.plugins.android.application) apply false
```

File: app/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        targetSdk = 33
+        versionCode = 25
+        versionName = "0.10.1"
+
+        buildConfigField("String", "USER_AGENT_STRING", "\"ReadYou/${'$'}{versionName}(${versionCode})\"")
+
+        testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+        vectorDrawables.useSupportLibrary = true
+
+        javaCompileOptions {
```
Comment (JunkFood02) [inline app/build.gradle.kts:43]: Though the migration of Room compiler is a non-goal, we should also migrate to `Room Gradle Plugin` in this PR.

Reference: https://developer.android.com/jetpack/androidx/releases/room

**Review by JunkFood02 — CHANGES_REQUESTED** [review]

**Comment by aualbert** [comment]
It should be done now.

File: app/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,173 @@
+import java.util.Properties
+import java.io.FileInputStream
+
+plugins {
+    id("com.android.application")
```
Comment (JunkFood02) [inline app/build.gradle.kts:5]: `alias` here too 😉

Comment (aualbert) [reply app/build.gradle.kts:5]: Ah ok I see!

Comment (aualbert) [reply app/build.gradle.kts]: I've replaced most of the `id` by `alias` but I do not know how to do it for hilt without breaking everything. I get strange errors about `android` being undefined. Could you have a look at it? Thanks.

File: app/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    packaging {
+        resources.excludes.add("/META-INF/{AL2.0,LGPL2.1}")
+    }
+    androidResources {
+        generateLocaleConfig = true
+    }
+    namespace = "me.ash.reader"
+}
+
+dependencies {
```
Comment (JunkFood02) [inline app/build.gradle.kts:113]: we have divided these dependencies into several groups before (separated with empty lines), preserve such structure could make this section look not that teriffying for newcomers

File: app/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+    implementation(libs.retrofit)
+    implementation(libs.retrofitGson)
+    implementation(libs.profileinstaller)
+    implementation(libs.workRuntimeKtx)
+    implementation(libs.datastorePreferences)
+    implementation(libs.roomPaging)
+    implementation(libs.roomCommon)
+    implementation(libs.roomKtx)
+    ksp(libs.roomCompiler)
+    implementation(libs.pagingCommonKtx)
```
Comment (JunkFood02) [inline app/build.gradle.kts:137]: Btw `libs.xxxxxx` seems not very well-structure, formats like `libs.coil.svg` and `libs.coil.compose` are preferred

File: gradle/libs.versions.toml [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+mockitoCore = { group = "org.mockito", name = "mockito-core", version.ref = "mockitoCore" }
+mockitoJUnitJupiter = { group = "org.mockito", name = "mockito-junit-jupiter", version.ref = "mockitoJUnitJupiter" }
+mockitoKotlin = { group = "org.mockito.kotlin", name = "mockito-kotlin", version.ref = "mockitoKotlin" }
+
+[plugins]
+android-application = { id = "com.android.application", version.ref = "androidGradlePlugin" }
+android-library = { id = "com.android.library", version.ref = "androidGradlePlugin"}
+kotlin-android = { id = "org.jetbrains.kotlin.android", version.ref = "kotlin" }
+ksp = { id = "com.google.devtools.ksp", version.ref = "ksp" }
+aboutlibraries-plugin = { id = "com.mikepenz.aboutlibraries.plugin", version.ref = "aboutLibsRelease" }
```
Comment (JunkFood02) [inline gradle/libs.versions.toml:104]: rename this to`aboutlibraries` to align with rest of the plugins
`alias(libs.plugins.aboutlibraries)`

File: app/build.gradle.kts (hunk 5 above)
Comment (JunkFood02) [reply app/build.gradle.kts:137]: reference: https://github.com/android/nowinandroid/blob/main/gradle/libs.versions.toml

File: gradle/libs.versions.toml [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+[libraries]
+aboutLibrariesCore = { group = "com.mikepenz", name = "aboutlibraries-core", version.ref = "aboutLibsRelease" }
+aboutLibrariesComposeM3 = { group = "com.mikepenz", name = "aboutlibraries-compose-m3", version.ref = "aboutLibsRelease" }
+composeHtml = { group = "com.github.ireward", name = "compose-html", version.ref = "composeHtml" }
+androidSVG = { group = "com.caverock", name = "androidsvg-aar", version.ref = "androidSVG" }
+opmlParser = { group = "be.ceau", name = "opml-parser", version.ref = "opmlParser" }
+readability4j = { group = "net.dankito.readability4j", name = "readability4j", version.ref = "readability4j" }
+rome = { group = "com.rometools", name = "rome", version.ref = "rome" }
+coilBase = { group = "io.coil-kt", name = "coil-base", version.ref = "coil" }
```
Comment (JunkFood02) [inline gradle/libs.versions.toml:48]: ditto, `coil-base`, `coil-compose` and `coil-svg`...

**Review by JunkFood02 — CHANGES_REQUESTED** [review]
Sorry for the late response, here're some other suggestions for revision 🫰

File: app/build.gradle.kts [hunk 8]
```diff
@@ -0,0 +1,172 @@
+import java.util.Properties
+import java.io.FileInputStream
+
+plugins {
+    kotlin("android")
```
Comment (JunkFood02) [inline app/build.gradle.kts:5]:
```suggestion
alias(libs.plugins.kotlin.android)
```

**Comment by aualbert** [comment]
Done. Do you see something else we could change?

**Review by JunkFood02 — APPROVED** [review]
LGTM, thx!

**Pull request merged**